### PR TITLE
Fix frequency processing for trips per hour histograms

### LINKED ISSUE #472: Frequency trips not represented correctly in trips per hour histograms
## Observed behavior

In the trips per hour histograms (routes and patterns) on the QA page, trips with frequencies are not being processed correctly -- they all end up in the hour bucket for whatever time the first stop_time for the trips is assigned.

image

## Expected behavior

The trip processing should be taking into account the frequency period and headway for any frequency trips.

## Steps to reproduce the problem

Load a feed with frequency trips, e.g., https://s3.amazonaws.com/setravi/GTFS_SEMOVI_CDMX.zip

## Notes on configuration and files required to reproduce the issue

N/A

### PR DESCRIPTION
**Pull request by landonreed**
### Checklist

- [x] Appropriate branch selected _(all PRs must first be merged to `dev` before they can be merged to `master`)_
- [x] Any modified or new methods or classes have helpful JSDoc and code is thoroughly commented
- [x] The description lists all applicable issues this PR seeks to resolve
- [x] The description lists any configuration setting(s) that differ from the default settings
- [x] All tests and CI builds passing

### Description

Fix #472

### CODE REVIEW

File: lib/manager/selectors/index.js [hunk 1]
```diff
@@ -73,46 +79,73 @@ export const getPatternData: AppState => Array<PatternRowData> = createSelector(
...
+  const patternLookup = {}
+  route.trips.forEach(trip => { patternLookup[trip.pattern_id] = true })
+  return Object.keys(patternLookup).length
+}
+
+/**
+ * Calculate trips per hour for an array of trips.
+ */
+function calculateTripsPerHour (trips: Array<ValidationTrip>): Array<number> {
+  const tripsPerHour = Array(28).fill(0)
```
Comment (evansiroky) [inline lib/manager/selectors/index.js]: Would be nice to have a comment what the 28 magic number means.

File: lib/manager/selectors/__tests__/index.js [hunk 2]
```diff
@@ -157,6 +159,7 @@ describe('manager > selectors >', () => {
             ],
             trips: [
               {
+                frequencies: [],
```
Comment (evansiroky) [inline lib/manager/selectors/__tests__/index.js]: Would be nice to either modify this test case or have another one for when frequencies do exist.

**Comment by landonreed** [comment]
Fixed the comment, but may need to leave the enhanced test for another time.

**Pull request merged**